AS-215 version in build artifact file name

bhiln opened the pull request:
Add AP version, latest mttr git tag in commit history, and git commit to `.apj` and `.elf` build artifact files.
![image](https://user-images.githubusercontent.com/2285455/213787373-ea67edac-21e4-4113-8d42-8f5e80f3a1e9.png)

bhiln committed:
AS-215 Add: AP version, mttr git tag, mttr git hash

bhiln committed:
AS-215 Fix: get latest mttr-* git tag in branch history

bhiln committed:
AS-215 Fix: only look at branch history for latest tag

bhiln committed:
AS-215 Add: fetch tags before query

bhiln committed:
AS-215 Fix: git fetch don't recurse submodules

bhiln committed:
AS-215 Update: different approach to getting latest tag

bhiln committed:
AS-215 Debug: only run tag code

bhiln committed:
Revert "AS-215 Debug: only run tag code"

This reverts commit 3272b4b7b6488c78db5b14871c0d21b48bb14195.

bhiln commented:
Check hash, this may be the wrong hash

robkarol commented on Tools/scripts/mttr-build-ci.sh:
```diff
@@ -26,8 +26,21 @@ echo $PATH
+MTTR_GIT_TAG=$(echo $(git tag -l "mttr-*" --sort=-committerdate | head -1) | tr - _)
+
+# Get mttr git hash
+MTTR_GIT_HASH=$(git rev-parse --short=7 HEAD)
```
When this gets run is very important. It must be run after the commit gets merged to the main for this to work. If you can run a test that shows the hash is correct we can leave this in. Otherwise it will cause way more confusion than its worth.

bhiln replied:
You are right. I ensured that it worked well on my pc, but I'm not sure it was right from circleci. I'm looking into this.

bhiln replied:
@robkarol Here's the latest push with correct hash in the files: https://github.com/matternet/ardupilot/actions/runs/4008770308
<img width="769" alt="image" src="https://user-images.githubusercontent.com/2285455/214675715-c5273fce-8687-4b7b-983a-6c9ebde46abf.png">
<img width="538" alt="image" src="https://user-images.githubusercontent.com/2285455/214675649-d7765b48-9d21-4dd4-9701-fe02005178a7.png">

robkarol commented:
This change may not be needed depending on a few conversations I have later this week. Lets hold off on merging it for now, also pending the test I suggested in the review.

bhiln committed:
AS-215 Remove: don't run github actions on PR
github actions already runs on push using the correct commit hash. This covers push, PR, and merges.
